### PR TITLE
feat(pie-icons-vue): DSW-684 Update compilation settings

### PR DESCRIPTION
**Pull request by fernandofranca**
## pie-icons-vue

- [Changed] - Updated build settings and removed redundant steps
- [Changed] - Updated bundling settings
- [Removed] - Redundant icons files
- [Changed] - Align scripts with pie-icons-react
- [Changed] - Test file and added ESLint rule for files generated during build

## pie-icons-react

- [Changed] - Align tests with `pie-icons-vue` and replace Jest with Vitest
- [Changed] - Align scripts with `pie-icons-vue`
- [Changed] - Build script to avoid race condition while writing index file
- [Changed] - Added `esm` folder to `.eslintignore`

## pie-monorepo

- [Changed] - Shared `@rollup/plugin-node-resolve` within the monorepo

### CODE REVIEW

File: packages/tools/pie-icons-react/scripts/build.mjs [hunk 1]
```diff
@@ -62,6 +55,9 @@ async function build () {
     // check that the /icons directory exists, if not create it
     await checkDirExists(ICONS_DIR);
 
+    // open a write stream to index.tsx
+    const indexFile = fs.createWriteStream(filePath);
```
Comment (fernandofranca) [inline packages/tools/pie-icons-react/scripts/build.mjs:59]: The write stream should be created after the directory is created, in the previous line.

File: packages/tools/pie-icons-vue/test/IconAlcohol.test.js [hunk 2]
```diff
@@ -9,7 +9,8 @@ import {
 // eslint-disable-next-line import/no-unresolved
 import { sizeToValueMap } from '@justeattakeaway/pie-icons-configs';
 
-import IconAlcohol from '../icons/IconAlcohol';
+// eslint-disable-next-line import/no-unresolved, import/extensions
+import IconAlcohol from '../esm/IconAlcohol';
```
Comment (fernandofranca) [inline packages/tools/pie-icons-vue/test/IconAlcohol.test.js:13]: `esm` files are cached when Turborepo runs the `build` script.

File: packages/tools/pie-icons-vue/test/IconAlcoholLarge.test.js [hunk 3]
```diff
@@ -6,7 +6,8 @@ import {
     vi,
 } from 'vitest';
 
-import IconAlcoholLarge from '../icons/IconAlcoholLarge';
+// eslint-disable-next-line import/no-unresolved, import/extensions
+import IconAlcoholLarge from '../esm/IconAlcoholLarge';
```
Comment (raoufswe) [inline packages/tools/pie-icons-vue/test/IconAlcoholLarge.test.js:10]: does this mean consumers will also need to update their imports to use esm?

**Review by raoufswe — APPROVED** [review]

File: packages/tools/pie-icons-vue/test/IconAlcoholLarge.test.js (hunk 3 above)
Comment (fernandofranca) [reply packages/tools/pie-icons-vue/test/IconAlcoholLarge.test.js:10]: I believe not. The package contains only `dist` and `esm`.

**Review by xander-marjoram — APPROVED** [review]

**Review by raoufswe — APPROVED** [review]

**Pull request merged**